cmd/xl-v1-healing.go

```diff
@@ -667,6 +667,19 @@ func (xl xlObjects) HealObject(ctx context.Context, bucket, object string, dryRu
 	// Read metadata files from all the disks
 	partsMetadata, errs := readAllXLMetadata(healCtx, storageDisks, bucket, object)
 
+	// Check if the object is corrupted, if yes and user requested
+	// remove we simply delete it from namespace, otherwise we
+	if m, ok := xl.isObjectCorrupted(partsMetadata, errs); ok {
```

Comment:
I am not sure I understand what `isObjectCorrupted` is doing - it looks like:

```go
func (xl xlObjects) isObjectCorrupted(metaArr []xlMetaV1, errs []error) (validMeta xlMetaV1, ok bool) {
	// We can consider an object data not reliable
	// when xl.json is not found in read quorum disks.
	var notFoundXLJSON, corruptedXLJSON int
	for _, readErr := range errs {
		if readErr == errFileNotFound {
			notFoundXLJSON++
		}
	}
	for _, readErr := range errs {
		if readErr == errCorruptedFormat {
			corruptedXLJSON++
		}
	}

	for _, m := range metaArr {
		if !m.IsValid() {
			continue
		}
		validMeta = m
		break
	}

	// Return if the object is indeed corrupted.
	return validMeta, len(xl.getDisks())-notFoundXLJSON < validMeta.Erasure.DataBlocks || len(xl.getDisks()) == corruptedXLJSON
}
```

I would think object is corrupted when `len(xl.getDisks()) - notFoundXLJSON - corruptedXLJSON < numDataBlocks` - the condition used in the code looks wrong to me.